vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridSingleSelectionIT.java

```diff
@@ -85,4 +85,25 @@ public void checkDeselectionDisallowedInitially() {
         Assert.assertTrue("Row 1 was still selected after de-selecting it.",
                 !grid.getRow(1).isSelected());
     }
+
+    @Test
+    public void gridSetItemsAndSelectItemAndSetItemAfterSetDeselectAllowedFalse() {
+        open();
+
+        // De-selection is allowed(deselectAllowed is false) and then setting items for grid
```

Comment:
Grid de-selection is not allowed at first (it's disallowed initially at `setDeselectAllowedAndSetItems`)

Reply:
correct cmt